src/librustc_trans_utils/trans_crate.rs

```diff
@@ -77,179 +77,28 @@ pub trait TransCrate {
 #[macro_export]
 macro_rules! hot_pluggable_trans_crate {
     (|$sess:ident| { $body:expr }) => {
-        use $crate::__rustc::ty::maps::Providers;
         #[no_mangle]
-        pub extern "C" fn __rustc_backend_new($sess: &Session) -> *mut Box<TransCrate> {
-            let trans_crate = { $body };
-            Box::into_raw(Box::new(trans_crate))
+        pub fn __rustc_backend_new($sess: &Session) -> Box<TransCrate> {
+            { $body }
```

Comment:
You don't need the braces here - also, do we really need a macro now?
I think the name should be `__rustc_codegen_backend`.